### PR TITLE
Inserting/deleting items from middle of a brickCollectionView

### PR DESCRIPTION
**Pull request by ethan-riback**
Items can now be inserted and deleted from anywhere in the brickCollectionView and the collectionView will be responsive to the change.

### CODE REVIEW

File: Example/Source/Bricks/Custom/SegmentHeaderBrick.swift [hunk 1]
```diff
@@ -40,6 +42,12 @@ class SegmentHeaderBrickCell: BrickCell, Bricklike {
 protocol SegmentHeaderBrickDataSource: class {
     var titles: [String] { get }
     var selectedSegmentIndex: Int { get }
+
+    func configure(cell: SegmentHeaderBrickCell)
+}
+
+extension SegmentHeaderBrickDataSource {
+    func configure(cell: SegmentHeaderBrickCell) {/*Optional*/}
```
Comment (rubencagnie) [inline Example/Source/Bricks/Custom/SegmentHeaderBrick.swift]: This is an old brick. Ideally, we should use `GenericBrick<UISegementedControl>`...

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift [hunk 2]
```diff
@@ -21,23 +23,39 @@ class InsertBrickViewController: BrickApp.BaseBrickController, HasTitle {
...
         }
     }
 
-    var titles: [String] {
-        return ["No animation", "From top", "From bottom"]
+    var appearTitles: [String] {
+        return ["No animation", "From top", "From bottom", "Fade in"]
+    }
+
+    var insertToIndex: Int {
```
Comment (rubencagnie) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: This variable name is a bit confusing, feels more like a function name. Can you call it more `indexToInsertTo`?

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift [hunk 3]
```diff
@@ -49,47 +67,38 @@ class InsertBrickViewController: BrickApp.BaseBrickController, HasTitle {
         self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(InsertBrickViewController.insertBrick))
 
         let section = BrickSection(Section, bricks: [
-            SegmentHeaderBrick(dataSource: self, delegate: self),
-            LabelBrick(BrickIdentifiers.repeatLabel, backgroundColor: UIColor.lightGray, dataSource: self)
+            BrickSection(bricks: [SegmentHeaderBrick("AppearSegmentHeaderBrick", dataSource: self, delegate: self),
```
Comment (rubencagnie) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: Can you make the string into constants?
Can you also align the bricks, so it's a bit more readable?
```
        let section = BrickSection(Section, bricks: [
            BrickSection(bricks: [
                        SegmentHeaderBrick("AppearSegmentHeaderBrick", dataSource: self, delegate: self),
                        SegmentHeaderBrick("InsertSegmentHeaderBrick", dataSource: self, delegate: self)
            ]),
            LabelBrick(BrickIdentifiers.repeatLabel, height: .fixed(size: 37), backgroundColor: UIColor.lightGray, dataSource: self)
            ], inset: 10, edgeInsets: UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5))
```

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift [hunk 4]
```diff
@@ -49,47 +67,38 @@ class InsertBrickViewController: BrickApp.BaseBrickController, HasTitle {
...
-        numberOfLabels += 1
-        updateRepeatCounts()
+        let newElement = "BRICK \(data.count + 1)"
+        let index = insertToIndex
+        if selectedInsertSegmentIndex == 2 {
+            data.append(newElement)
+        } else {
+            data.insert(newElement, at: index)
+        }
+        updateRepeatCounts(fixedInsertedIndexPaths: [IndexPath(item: index + 1, section: 1)])
```
Comment (rubencagnie) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: BrickKit is designed so the consumer doesn't need to know about indexPaths, so this concerns me a bit. I probably want the API to be a bit more in the trend of
```
insertItem(for identifier: String, atIndex index: Int)
```

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift [hunk 5]
```diff
@@ -12,7 +12,9 @@ import BrickKit
 private let Section = "Section"
 
 class InsertBrickViewController: BrickApp.BaseBrickController, HasTitle {
-    
+
+    var data: [String] = ["Brick 1"]
```
Comment (rubencagnie) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: The repeated bricks have the title `let newElement = "BRICK \(data.count + 1)"`. So this should be upper case String

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift [hunk 6]
```diff
@@ -21,23 +23,39 @@ class InsertBrickViewController: BrickApp.BaseBrickController, HasTitle {
...
             default: appearBehavior = nil
             }
             self.layout.appearBehavior = appearBehavior
         }
     }
 
-    var titles: [String] {
-        return ["No animation", "From top", "From bottom"]
+    var appearTitles: [String] {
+        return ["No animation", "From top", "From bottom", "Fade in"]
```
Comment (rubencagnie) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: This doesn't look good on an iPhone 5(S). Can we abbreviate these a bit more?

File: Example/Source/Bricks/Custom/SegmentHeaderBrick.swift (hunk 1 above)
Comment (ethan-riback) [inline Example/Source/Bricks/Custom/SegmentHeaderBrick.swift]: Changed this to generic brick

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift (hunk 3 above)
Comment (ethan-riback) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: fixed

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift (hunk 4 above)
Comment (ethan-riback) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: Fixed this

File: Example/Source/Examples/Interactive/InsertBrickViewController.swift (hunk 5 above)
Comment (ethan-riback) [inline Example/Source/Examples/Interactive/InsertBrickViewController.swift]: Fixed

**Pull request merged**